app/backend/approaches/chatreadretrieveread.py

```diff
@@ -126,11 +146,9 @@ async def run_until_final_call(
             few_shots=self.query_prompt_few_shots,
         )
 
-        chatgpt_args = {"deployment_id": self.chatgpt_deployment} if self.openai_host == "azure" else {}
-        chat_completion = await openai.ChatCompletion.acreate(
-            **chatgpt_args,
+        chat_completion: ChatCompletion = await self.openai_chat_client.chat.completions.create(
+            messages=messages,  # type: ignore
```

Comment:
What doesn't it like about messages?

Reply:
I've typed that. There's 1 remaining type: ignore that mypy is still complaining but I can't figure out why

Reply:
See https://github.com/Azure-Samples/azure-search-openai-demo/pull/1017/commits/a9ef636438c0db11d2de659e5ebaacca74217316